packages/react-hooks/src/ConnectionProvider.tsx

```diff
@@ -15,21 +15,57 @@ import {
   addRecord,
 } from './recordUtils'
 
+export type useConnectionsOptions = {
+  excludedTypes?: [ConnectionType | string]
+  connectionState?: DidExchangeState
+}
+
 const ConnectionContext = createContext<RecordsState<ConnectionRecord> | undefined>(undefined)
 
-export const useConnections = () => {
+/**
+ * This method retrieves the connection context for the current agent.
+ * From this you can access all connection records for the agent.
+ * @param options options for useConnections hook, lets us filter out specific types and limit states
+ * @returns a connection context containing information about the agents connections
+ */
+export const useConnections = (options: useConnectionsOptions = {}) => {
   const connectionContext = useContext(ConnectionContext)
-  if (!connectionContext) {
-    throw new Error('useConnections must be used within a ConnectionContextProvider')
-  }
-  return connectionContext
+
+  let connections = connectionContext?.records
+  
+  connections = useMemo(() => {
+    if (!connections) {
+      throw new Error('useConnections must be used within a ConnectionContextProvider')
+    }
+    // do not filter if not filter options are provided to save on a loop
+    if (!options.connectionState && !options.excludedTypes) return connections
+
+    return connections.filter((record: ConnectionRecord) => {
+      // By default we include this connection
+      
+      // Filter by state (if connectionState is defined)
+      if (options.connectionState !== record.state) return false
```

Comment:
This will always return false if you don't provide a connectionState to the options, while it should return true I think

```suggestion
      // Filter by state (if connectionState is defined)
      if (options.connectionState && options.connectionState !== record.state) return false
```

Reply:
Might be good to add some tests? As this seems like a tricky hook

Reply:
Agreed, though would it be okay if we added tests in a separate PR? I did at least run bifold with the current iteration after adding this check and it appears to be working as intended.